File: src/Web/ClientApp/src/app/components/doctors/stats.ts
export interface IStats {
  "rating": number,
  "countComments": number,
  "countLikes": number,
  "countDisLikes": number
}

export class Stats implements IStats {
  countComments: number;
  countDisLikes: number;
  countLikes: number;
  rating: number;

  static deserialize(data: any): Stats {
    const stats = new Stats()

    stats.countComments = data.countComments;
    stats.countDisLikes = data.countDisLikes;
    stats.countLikes = data.countLikes;
    stats.rating = data.rating;

    return stats;
  }
}

File: src/Domain/Entities/DoctorAggregate/Stats.cs
namespace DocHelper.Domain.Entities.DoctorAggregate
{
    public class Stats : BaseEntity
    {
        public double Rating { get; private set; }
        public int CountComments { get; private set; }
        public int CountLikes { get; private set; }
        public int CountDisLikes { get; private set; }

        public Doctor Doctor { get; private set; }
        public int DoctorId { get; set; }

        private Stats()
        {
        }
        
        public Stats(Doctor doctor)
        {
            Doctor = doctor;
        }

        public Stats(double rating, int countComments, int countLikes, int countDisLikes, Doctor doctor)
        {
            Rating = rating;
            CountComments = countComments;
            CountLikes = countLikes;
            CountDisLikes = countDisLikes;
            Doctor = doctor;
        }
    }
}